File: lib/views/main/main_tab_view.dart
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';
import 'package:le_dariel/services/auth_service.dart';
import 'package:le_dariel/services/acronym_manager.dart';
import 'package:le_dariel/services/acronym_access_service.dart';
import 'package:le_dariel/views/main/acronym_list_view.dart';
import 'package:le_dariel/views/main/alphabetical_index_view.dart';
import 'package:le_dariel/views/main/profile_view.dart';
import 'package:le_dariel/views/admin/admin_view.dart';
import 'package:le_dariel/models/user.dart';
import 'package:le_dariel/services/acronym_access_service.dart';

class MainTabView extends StatefulWidget {
  const MainTabView({super.key});

  @override
  State<MainTabView> createState() => _MainTabViewState();
}

class _MainTabViewState extends State<MainTabView> {
  int _selectedIndex = 0;

  @override
  Widget build(BuildContext context) {
    final authService = Provider.of<AuthService>(context);
    final acronymManager = Provider.of<AcronymManager>(context);
    final accessService = Provider.of<AcronymAccessService>(context);
    final isAdmin = authService.currentUser?.userType == UserType.administrateur;

    final tabs = <Widget>[
      AcronymListView(
        accessService: accessService,
        acronymManager: acronymManager,
      ),
      AlphabeticalIndexView(accessService: accessService),
      ProfileView(
        authService: authService,
        accessService: accessService,
      ),
    ];

    if (isAdmin) {
      tabs.add(AdminView(
        authService: authService,
        acronymManager: acronymManager,
      ));
    }

    return Scaffold(
      body: IndexedStack(
        index: _selectedIndex,
        children: tabs,
      ),
      bottomNavigationBar: NavigationBar(
        selectedIndex: _selectedIndex,
        onDestinationSelected: (int index) {
          setState(() {
            _selectedIndex = index;
          });
        },
        destinations: [
          const NavigationDestination(
            icon: Icon(Icons.book_outlined),
            selectedIcon: Icon(Icons.book),
            label: 'Dictionnaire',
          ),
          const NavigationDestination(
            icon: Icon(Icons.list_outlined),
            selectedIcon: Icon(Icons.list),
            label: 'Index',
          ),
          const NavigationDestination(
            icon: Icon(Icons.person_outline),
            selectedIcon: Icon(Icons.person),
            label: 'Profil',
          ),
          if (isAdmin)
            const NavigationDestination(
              icon: Icon(Icons.settings_outlined),
              selectedIcon: Icon(Icons.settings),
              label: 'Admin',
            ),
        ],
      ),
    );
  }
}

File: Le_DARIEL/Views/Main/MainTabView.swift
//
//  MainTabView.swift
//  Le_DARIEL
//
//  Created by FAEI971 on 23/07/2025.
//

import SwiftUI

struct MainTabView: View {
    @StateObject private var authService = AuthService()
    @StateObject private var acronymManager = AcronymManager()
    @StateObject private var accessService: AcronymAccessService
    
    @State private var selectedTab = 0
    
    init() {
        let auth = AuthService()
        let manager = AcronymManager()
        _authService = StateObject(wrappedValue: auth)
        _acronymManager = StateObject(wrappedValue: manager)
        _accessService = StateObject(wrappedValue: AcronymAccessService(authService: auth, acronymManager: manager))
    }
    
    var body: some View {
        Group {
            if authService.isAuthenticated {
                TabView(selection: $selectedTab) {
                    // Onglet Dictionnaire
                    AcronymListView(accessService: accessService, acronymManager: acronymManager)
                        .tabItem {
                            Label("Dictionnaire", systemImage: "book.fill")
                        }
                        .tag(0)
                    
                    // Onglet Index (A-Z)
                    AlphabeticalIndexView(accessService: accessService)
                        .tabItem {
                            Label("Index", systemImage: "list.bullet")
                        }
                        .tag(1)
                    
                    // Onglet Profil
                    ProfileView(authService: authService, accessService: accessService)
                        .tabItem {
                            Label("Profil", systemImage: "person.fill")
                        }
                        .tag(2)
                    
                    // Onglet Admin (seulement pour les admins)
                    if authService.currentUser?.userType == .administrateur {
                        AdminView(authService: authService, acronymManager: acronymManager)
                            .tabItem {
                                Label("Admin", systemImage: "gearshape.fill")
                            }
                            .tag(3)
                    }
                }
                .accentColor(DARIELTheme.accentRed)
            } else {
                LoginView()
            }
        }
    }
}





